### PR TITLE
Transform Vite HTML imports for Civet

### PR DESCRIPTION
**Pull request by Mokshit06**
This PR transforms the index.html file when being used with vite to add `outExt?transform` to the end of the import so that it gets transformed by the unplugin

### CODE REVIEW

File: integration/unplugin/src/index.ts [hunk 1]
```diff
@@ -252,6 +256,20 @@ const civetUnplugin = createUnplugin((options: PluginOptions = {}) => {
 
         return null;
       },
+      async transformIndexHtml(html) {
+        const cheerio = await import('cheerio');
+        const $ = cheerio.load(html);
```
Comment (edemaine) [inline integration/unplugin/src/index.ts:261]: Can we avoid the `cheerio` dependency? Civet aims to have a minimum number of dependencies. I think in this case we could probably write a careful regex...

File: integration/unplugin/src/index.ts [hunk 2]
```diff
@@ -252,6 +256,20 @@ const civetUnplugin = createUnplugin((options: PluginOptions = {}) => {
 
         return null;
       },
+      async transformIndexHtml(html) {
```
Comment (edemaine) [inline integration/unplugin/src/index.ts:259]: Should we do this only in dev mode? Or is it fine to have in both?

Comment (Mokshit06) [reply integration/unplugin/src/index.ts:259]: Should be fine in both

**Comment by edemaine** [comment]
I replaced the cheerio dependency with what I think is a reasonable regex HTML parser (for the purpose of detecting `<script>` tags). However, I can't get this to work on Linux or Windows. I get this error:

```
[vite] Internal server error: ENOENT: no such file or directory, open '/src/main.civet'
```

Is this some form of #774? The path is relative in the HTML file, though.

It also seems that, by the time the id is passed to `resolveId`, it has lost the `?transform` suffix. So I think the code might be getting confused because of that... Did you actually get this working in some test environment?

**Comment by Mokshit06** [comment]
Strange, I did have it working on MacOS. Let me check once again

**Comment by Mokshit06** [comment]
This change should no longer be required with #786. Also I'm pretty sure I've come across a bug either in Unplugin or Vite regarding how the paths are passed in from `resolveId` to `load`. Will investigate further to check which package is causing it

**Review by edemaine — APPROVED** [review]
With #786 merged in, this now works for me. Great!

**Pull request merged**